### PR TITLE
[Enhance] beautify the print result of the registry

### PR DESCRIPTION
**Pull request by Eiuyc**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

For better debug usages

## Modification

- requirements/runtime.txt
    - add tabulate requirement
- mmengine\registry\registry.py
    - use tabulate to format the repr str
- tests\test_registry\test_registry.py
    - modify the repr test

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: requirements/runtime.txt [hunk 1]
```diff
@@ -4,5 +4,6 @@ numpy
 pyyaml
 regex;sys_platform=='win32'
 rich
+tabulate
```
Comment (zhouzaida) [inline requirements/runtime.txt:7]: Can we avoid adding the new dependency `tabulate` by using [the table of rich](https://rich.readthedocs.io/en/latest/tables.html)?

Comment (Eiuyc) [reply requirements/runtime.txt:7]: Okk, I'll do it.

Comment (Eiuyc) [reply requirements/runtime.txt]: The code has been updated, please review.

**Comment by zhouzaida** [comment]
image
image

**Comment by zhouzaida** [comment]
image
image

**Comment by Eiuyc** [comment]
> Hi @Eiuyc , thanks for your contribution. This result is beautiful. image image

Thank you! I'm happy to hear that.

File: mmengine/registry/registry.py [hunk 2]
```diff
@@ -120,10 +124,25 @@ def __contains__(self, key):
...
+        table = Table(title=f'Registry of {self._name}')
+        table.add_column('Names', justify='right', style='cyan')
+        table.add_column('Objects', justify='right', style='green')
+
+        # In table header, len('Names') == 5 and len('Objects') == 7
+        max_name_width = 5
+        max_obj_width = 7
+        for name, obj in self._module_dict.items():
+            max_name_width = max(max_name_width, len(name))
+            max_obj_width = max(max_obj_width, len(str(obj)))
```
Comment (zhouzaida) [inline mmengine/registry/registry.py:136]: If we do not set the `table_width`, can it work as expected?

**Comment by ZwwWayne** [comment]
Overall, this PR looks good to me, and the printed results are pretty.

File: mmengine/registry/registry.py (hunk 2 above)
Comment (Eiuyc) [reply mmengine/registry/registry.py:136]: Snipaste_2023-02-13_17-12-34
The `rich` will automatically hide some chars, and broke table struct, which seems not good.

Comment (zhouzaida) [reply mmengine/registry/registry.py:136]: If table_width is set, the longest string will be splitter two lines.

image

If table_width is not set, it works well.

image

Comment (Eiuyc) [reply mmengine/registry/registry.py:136]: @zhouzaida Thank you, you're right. The previous img I attached was “a test unit captured” std output, which might be different from interactive terminal. I'll fix the width issue ASAP.

**Comment by Eiuyc** [comment]
Hi, the code has been updated, please review.

File: mmengine/registry/registry.py (hunk 2 above)
Comment (Eiuyc) [reply mmengine/registry/registry.py]: image
That's how print result looks now. It turns out "setting width" is not necessary :)

**Comment by zhouzaida** [comment]
> Hi, the code has been updated, please review.

Look good to me. Thanks for your contribution.

**Review by zhouzaida — APPROVED** [review]

**Comment by jason102811** [comment]
Hi @Eiuyc  ！First of all, we want to express our gratitude for your significant PR in the MMEngine . Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**